src/mgclient.c

```diff
@@ -441,8 +452,15 @@ static int init_tcp_connection(const mg_session_params *params, int *sockfd,
     abort();
   }
   if (getaddrinfo_status != 0) {
+#ifdef __EMSCRIPTEN__
+    mg_session_set_error(session, "getaddrinfo failed: %s",
+                         "Replace with error");
+    // Not supported by emscripten:
+    // gai_strerror(getaddrinfo_status));
+#else
     mg_session_set_error(session, "getaddrinfo failed: %s",
                          gai_strerror(getaddrinfo_status));
+#endif
```

Comment:
```suggestion
#ifdef __EMSCRIPTEN__
    const char *error = 
    mg_session_set_error(session, "getaddrinfo failed");
    // Not supported by emscripten:
    // gai_strerror(getaddrinfo_status));
#else
    mg_session_set_error(session, "getaddrinfo failed: %s",
                         gai_strerror(getaddrinfo_status));
#endif
```

Reply:
Why receive error in one case and not in other?